tests/rptest/tests/many_clients_test.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2022 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.clients.types import TopicSpec
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.redpanda import ResourceSettings, RESTART_LOG_ALLOW_LIST
+from rptest.services.cluster import cluster
+from rptest.services.rpk_consumer import RpkConsumer
+
+from ducktape.utils.util import wait_until
+from ducktape.cluster.cluster_spec import ClusterSpec
+
+from rptest.services.producer_swarm import ProducerSwarm
+
+resource_settings = ResourceSettings(
+    num_cpus=2,
+
+    # Set a low memory size, such that there is only ~100k of memory available
+    # for dealing with each client.
+    memory_mb=384,
+
+    # Test nodes and developer workstations may have slow fsync.  For this test
+    # we need things to be consistently fast, so disable fsync (this test
+    # has nothing to do with verifying the correctness of the storage layer)
+    bypass_fsync=True)
+
+
+class ManyClientsTest(RedpandaTest):
+    def __init__(self, *args, **kwargs):
+        # We will send huge numbers of messages, so tune down the log verbosity
+        # as this is just a "did we stay up?" test
+        kwargs['log_level'] = "info"
+        kwargs['resource_settings'] = resource_settings
+        super().__init__(*args, **kwargs)
+
+    @cluster(num_nodes=6)
+    def test_many_clients(self):
+        """
+        Check that redpanda remains stable under higher numbers of clients
+        than usual.
+        """
+
+        if self.debug_mode:
+            self.logger.info("Skipping test, not suitable for debug mode")
+            # A phony allocation to satisfy the check that we used all
+            # the nodes in our `num_nodes` value.
+            alloc_nodes = self.test_context.cluster.alloc(
+                ClusterSpec.simple_linux(3))
+            self.test_context.cluster.free(alloc_nodes)
+            return
+
+        PARTITION_COUNT = 100
+        PRODUCER_COUNT = 4000
+        TOPIC_NAME = "manyclients"
+        RECORDS_PER_PRODUCER = 1000
```

Comment:
are these settings something we're likely to adjust later? IIRC from running conn_test a while back I had some configurations that took more than 30 seconds to run. not sure what is 'good enough'?